projects/plugins/jetpack/modules/search/instant-search/lib/constants.js

```diff
@@ -14,6 +14,7 @@ export const RESULT_FORMAT_MINIMAL = 'minimal';
 export const RESULT_FORMAT_PRODUCT = 'product';
 export const MINUTE_IN_MILLISECONDS = 60 * 1000;
 export const RELEVANCE_SORT_KEY = 'relevance';
+export const DEBOUNCED_TIME_TO_SET_QUERY_MILLISECONDS = 1000;
```

Comment:
It'd be great if we could match this delay to the median response time for our API. I can't name the figure off the top of my head, but it's likely around 500ms or so?

Reply:
The direct call to our `public-api` is pretty fast, which is like `200ms`, while the forwarded API (private sites) response time is like `600ms`. But I guess, the delay would be more related to how long a user might pause after he's done with a full query. After user finishes entering, he would spent a couple of seconds looking at the results. So I'm not worried that 1 second is too long.

Tried 500ms tho, from testing, there are still quite a few interim URLs generate in the middle of the input. So I guess we'd stick to 1 second unless you still think otherwise.